src/Land/Model/soil_water.jl

```diff
@@ -123,44 +124,105 @@ end
 
 """
     get_water_content(
+        water::SoilWaterModel,
         aux::Vars,
         state::Vars,
-        t::Real,
-        water::SoilWaterModel,
+        t::Real
     )
 
 Return the moisture variables for the balance law soil water model.
 """
 function get_water_content(
+    water::SoilWaterModel,
     aux::Vars,
     state::Vars,
     t::Real,
-    water::SoilWaterModel,
 )
-    return state.soil.water.ϑ_l, state.soil.water.θ_ice
+    FT = eltype(state)
+    return FT(state.soil.water.ϑ_l), FT(state.soil.water.θ_ice)
 end
 
 
 
 """
     get_water_content(
+        water::PrescribedWaterModel,
         aux::Vars,
         state::Vars,
-        t::Real,
-        water::PrescribedWaterModel,
+        t::Real
     )
 
 Return the moisture variables for the prescribed soil water model.
 """
 function get_water_content(
+    water::PrescribedWaterModel,
     aux::Vars,
     state::Vars,
     t::Real,
+)
+    FT = eltype(aux)
+    ϑ_l = water.ϑ_l(aux, t)
+    θ_ice = water.θ_ice(aux, t)
+    return FT(ϑ_l), FT(θ_ice)
+end
+
+
+
+
+"""
+    get_initial_water_content(
+        aux::Vars,
+        t::Real,
+        water::SoilWaterModel,
+    )
+
+Return the moisture variables for the balance law soil water model, not from state.
+"""
+#function get_initial_water_content(
```

Comment:
we'll have to add this in in the next PR, but it is ok for now!

Reply:
oh actually I see that we arent using this function. We can see if we need it and if not we can eventually remove it.

Reply:
ok! didnt change anything